### ISSUE TITLE
Gradio Lite: Unable to connect to locally hosted Ollama server

### ISSUE THREAD

**Issue by saikatkumardey** [gradio-app/gradio#6048]
### Describe the bug

Hi,

I am trying to use Ollama server hosted locally from gradio lite. 

After running, I get the following error in the browser console:
```
requests.exceptions.ConnectionError: HTTPConnectionPool(host='127.0.0.1', port=11434): Max retries exceeded with url: /api/generate (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x74bae48>: Failed to establish a new connection: [Errno 50] Protocol not available'))
```

I tried via CURL and it works:
```
 curl -X POST http://localhost:11434/api/generate -d '{
  "model": "llama2",
  "prompt": "Why is the sky blue?"
}'
```

Any suggestions would be helpful!

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

```
<html>
	<head>
		<script type="module" crossorigin src="https://cdn.jsdelivr.net/npm/@gradio/lite/dist/lite.js"></script>
		<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@gradio/lite/dist/lite.css" />
	</head>
    <body>
		<gradio-lite>
                import random
                import requests
                import json
        
                url = "http://localhost:11434/api/generate"
        
                def ollama_response(query, history):
                    payload = {
                        "model": "llama-2",
                        "prompt": query
                    }
        
                    # Sending POST request and streaming the response
                    response = requests.post(url, json=payload)
        
                    # Iterate over the streaming response
                    for line in response.iter_lines():
                        if line:
                            parsed_line = json.loads(line.decode('utf-8'))
                            yield parsed_line["response"]
        
                import gradio as gr
                gr.ChatInterface(ollama_response).queue().launch()
		</gradio-lite>
	</body>
</html>
```

### System Info

```shell
I am using the latest version of @gradio/lite.
```

### Severity

Blocking usage of gradio

**Comment by OpenSpacesAndPlaces** [comment]
Noting ticket opened on Ollama
https://github.com/jmorganca/ollama/issues/1411

**Comment by abidlabs** [comment]
@whitphx do you now if Gradio/Lite can hit local URLs like this?

**Comment by whitphx** [comment]
@saikatkumardey Can you please try using [`pyodide.http.pyfetch()`](https://pyodide.org/en/stable/usage/api/python-api/http.html#pyodide.http.pyfetch) instead of `requests.post`, like
```python
import json
import pyodide.http


async def ollama_response(query, history):
	...
	response = await pyodide.http.pyfetch(url, method="POST", body=json.dumps(payload))
	text = await response.text()
	...
```

The `requests` library can't be used on Pyodide (Gradio-Lite) as [this doc](https://pyodide.org/en/stable/usage/faq.html#how-can-i-load-external-files-in-pyodide) explains.

We should have added this to our docs, sorry.

**Comment by whitphx** [comment]
This will also be fixed with #7302. Please try the workaround above until that.

**Comment by abidlabs** [comment]
Fixed via #7302!
